### ISSUE TITLE
sbs的cfg里，没有用到cricle loss？

### ISSUE THREAD

**Issue by elvindp** [JDAI-CV/fast-reid#591]
1. 在sbs的说明里，提到用circle loss的trick，但cfg里好像没有设置这个loss。不加的效果更好吗？
2. 此外，在NAIC20里，用到了cosface，请问一下，加了cosface可以提高效果吗？
3. 用sbs r50 ibn训练DukeMTMC的结果很差，rank-1在训练68个epoch达到89.18之后就开始下降，676个epoch后只剩83.66，中间在85徘徊了很久。这是正常的现象吗？

circle softmax就是circle loss……是我傻逼了……
但circle loss里的pairwise loss和circle softmax的区别是什么？什么时候用？

谢谢

**Comment by elvindp** [comment]
用sbs r50 ibn训练DukeMTMC的结果很差，rank-1在训练68个epoch达到89.18之后就开始下降，676个epoch后只剩83.66，中间在85徘徊了很久。这是正常的现象吗？

**Comment by BokyLiu** [comment]
https://github.com/JDAI-CV/fast-reid/issues/225#issuecomment-673911446

**Comment by L1aoXingyu** [comment]
circle softmax 和 pair-wise circle 可以具体看看 circle loss 那篇 paper，主要就是分类 loss 和度量学习 loss 的区别

**Comment by elvindp** [comment]
> circle softmax 和 pair-wise circle 可以具体看看 circle loss 那篇 paper，主要就是分类 loss 和度量学习 loss 的区别

明白了，所以，
一个输入时，是一组logits，和softmax里存放的矩阵乘
另一个是embeding pair。

**Comment by JoJoliking** [comment]
@elvindp @L1aoXingyu 请问以下，pair-wise circle的输入，embedding是Batch*Embedding Dim的Tensor。Target是对应什么呢？对应这个Batch个的ID号么？
